code/game/objects/items/storage/storage.dm

```diff
@@ -126,81 +171,91 @@
 		user.s_active = null
 	content_watchers -= user
 
-
+///Returns a list of every mob who has this storage open
 /obj/item/storage/proc/can_see_content()
 	var/list/lookers = list()
-	for(var/i in content_watchers)
-		var/mob/M = i
-		if(M.s_active == src && M.client)
-			lookers |= M
+	for(var/mob/watcher AS in content_watchers)
+		if(watcher.s_active == src && watcher.client)
+			lookers |= watcher
 		else
-			content_watchers -= M
+			content_watchers -= watcher
 	return lookers
 
+///Opens the view of this storage for the given mob
 /obj/item/storage/proc/open(mob/user)
-	if(!opened)
+	if(!(flags_storage & STORAGE_FLAG_OPENED))
 		orient2hud()
-		opened = 1
+		flags_storage |= STORAGE_FLAG_OPENED
 	if (use_sound)
 		playsound(src.loc, src.use_sound, 25, 1, 3)
 
 	if (user.s_active)
 		user.s_active.close(user)
 	show_to(user)
 
-
+///Closes the view of this storage for the given mob
 /obj/item/storage/proc/close(mob/user)
 	hide_from(user)
-
-
-//This proc draws out the inventory and places the items on it. tx and ty are the upper left tile and mx, my are the bottm right.
-//The numbers are calculated from the bottom-left The bottom-left slot being 1,1.
+/**
+ * This proc draws out the inventory and places the items on it. tx and ty are the upper left tile and mx, my are the bottm right.
+ * The numbers are calculated from the bottom-left The bottom-left slot being 1,1.
+ */
 /obj/item/storage/proc/orient_objs(tx, ty, mx, my)
 	var/cx = tx
 	var/cy = ty
 	boxes.screen_loc = "[tx]:,[ty] to [mx],[my]"
-	for(var/obj/O in src.contents)
-		O.screen_loc = "[cx],[cy]"
-		O.layer = ABOVE_HUD_LAYER
-		O.plane = ABOVE_HUD_PLANE
+	for(var/content in contents)
+		if(!isobj(content))
+			continue
+		var/obj/object = content
+		object.screen_loc = "[cx],[cy]"
+		object.layer = ABOVE_HUD_LAYER
+		object.plane = ABOVE_HUD_PLANE
```

Comment:
i feel storages should be able to handle atom/movable overall, imagine mobs inside storages! but you can leave that for anytime or someone else
```suggestion
	for(var/obj/content in contents)
		content.screen_loc = "[cx],[cy]"
		content.layer = ABOVE_HUD_LAYER
		content.plane = ABOVE_HUD_PLANE
```

Reply:
This sounds like something I'll do later after I recharge from this monster PR